ext/io/event/selector/kqueue.c

```diff
@@ -60,6 +64,9 @@ struct IO_Event_Selector_KQueue
 	int descriptor;
 	int blocked;
 	
+#ifndef EVFILT_USER
```

Comment:
I'd prefer if we have a feature test early in the file, e.g.

```c
#ifndef EVFILT_USER
#define IO_EVENT_SELECTOR_KQUEUE_USE_INTERRUPT
#endif
```

I'm not strongly attached to the specific name, howeve rhis helps to keep the feature independent of the reason for enabling it (there could be multiple reasons in the future or it might be controlled by `extconf.rb`.

Reply:
Yeah I wonder if OS X actually needs to "pre-wait" too.

I tested it with:
```ruby
require 'async'
Sync do
        100.times do
                fail unless `echo test` == "test\n"
                fail unless $?.success?
                warn $?.inspect
        end
end
```

It also needs the fixes in branch "fix-file-descriptor-race" to work reliably.

And right now it can fail (very rarely) due PID/FD conflicts causing errors when trying to disarm events of the wrong type.